c2rust-analyze/tests/filecheck/rewrite_nullable_box.rs

```diff
@@ -0,0 +1,35 @@
+//! Test cases for rewriting of nullable owned pointers.
+#![feature(rustc_private)]
+#![feature(register_tool)]
+#![register_tool(c2rust_analyze_test)]
+
+use std::mem;
+extern crate libc;
+
+extern "C" {
+    fn malloc(_: libc::c_ulong) -> *mut libc::c_void;
+    fn free(__ptr: *mut libc::c_void);
+}
+
+#[c2rust_analyze_test::skip_borrowck]
+// CHECK-LABEL: fn test_is_null(
+pub unsafe fn test_is_null(cond: bool) {
+    // CHECK: let mut ptr: core::option::Option<core::result::Result<std::boxed::Box<(i32)>,()>> =
```

Comment:
Why is it `Box<(i32)>`? Is that inner tuple needed? (or is it something else)

Reply:
That's not a tuple (1-tuples require a trailing comma, like `(i32,)`).  When reusing part of a type annotation as written in the original source code (in this case, the `i32` from `*mut i32`), we wrap the reused type in parentheses so we don't have to worry about precedence or other interactions with the surrounding rewrite.  In theory, these parentheses should be easy to detect and remove with an "unused parens" lint in the cases where they're not needed.